index.js

```diff
@@ -405,7 +405,22 @@ SendStream.prototype.onStatError = function onStatError (error) {
  */
 
 SendStream.prototype.isFresh = function isFresh () {
-  return fresh(this.req.headers, this.res._headers)
+  var resHeaders = this.res._headers || {}
+
+  // Support changes in node with how response headers are stored
+  var headerNames = this.res._headerNames || {}
+  var keys = Object.keys(resHeaders)
+  if (keys.length > 0 && headerNames[keys[0]] === undefined) {
```

Comment:
I suppose that in that case, this would just end up with all the values being `undefined` (or the second copy of the header if someone set the value to an array).